File: 백준/Silver/10845. 큐/큐.py
import sys

n = int(sys.stdin.readline())
que = []
s = 0

for i in range(n):
    order = sys.stdin.readline().strip()
    lenQue = len(que) - s
    
    if "push" in order:
        od = order.split()
        que.append(int(od[1]))

    elif order == "pop":
        if lenQue > 0:
            print(que[s])
            s += 1
        else:
            print(-1)

    elif order == "size":
        print(lenQue)

    elif order == "empty":
        if lenQue > 0:
            print(0)
        else:
            print(1)

    elif order == "front":
        if lenQue > 0:
            print(que[s])
        else:
            print(-1)

    elif order == "back":
        if lenQue > 0:
            print(que[len(que)-1])
        else:
            print(-1)

File: 백준/Silver/10845. 큐/큐.java

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;

public class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		Deque<Integer> dq = new ArrayDeque<>();
		StringBuilder sb = new StringBuilder();

		int n = Integer.parseInt(br.readLine());

		for (int i = 0; i < n; i++) {
			String order = br.readLine();
			switch (order) {
			case "front":
				sb.append((dq.peekFirst() != null ? dq.peekFirst() : -1) + "\n");
				break;
			case "back":
				sb.append((dq.peekLast() != null ? dq.peekLast() : -1) + "\n");
				break;
			case "size":
				sb.append(dq.size() + "\n");
				break;
			case "empty":
				sb.append((dq.isEmpty() ? 1 : 0) + "\n");
				break;
			case "pop":
				sb.append((dq.peekFirst() != null ? dq.pollFirst() : -1) + "\n");
				break;
			default:
				if (order.length() > 5 && order.split(" ")[0].equals("push")) {
					dq.offer(Integer.parseInt(order.split(" ")[1]));
				}
			}
		}

		System.out.println(sb);
	}
}